libunwind/src/AddressSpace.hpp

```diff
@@ -578,6 +578,11 @@ inline bool LocalAddressSpace::findUnwindSections(pint_t targetAddr,
 #ifdef __CHERI_PURE_CAPABILITY__
         check_same_type<__uintcap_t, decltype(pinfo->dlpi_addr)>();
         check_same_type<const Elf_Phdr *, decltype(pinfo->dlpi_phdr)>();
+
+        // Cannot use CTestSubset here because the dpli_addr perms are a strict
+        // subset that never includes execute and so won't match targetAddr
+        // which is always executable.
+        //
```

Comment:
You could use `__builtin_cheri_test_subset(__builtin_cheri_perms_and(cbdata->targetAddr, 0), pinfo->dlpi_addr)` to moot the permissions check?  But maybe that's too obfuscated. :)